components/admin_console/system_user_detail/system_user_detail.tsx

```diff
@@ -68,99 +91,100 @@ export default class SystemUserDetail extends React.PureComponent {
             customComponentWrapperClass: '',
             user: {
                 email: this.props.user.email,
-            },
+            } as UserProfile,
             addTeamOpen: false,
             refreshTeams: true,
+            error: '',
         };
     }
 
-    setTeamsData = (teams) => {
-        const teamIds = teams.map((team) => team.team_id);
+    setTeamsData = (teams: Record<string, any>): void => {
+        const teamIds = teams.map((team: GroupTeam) => team.team_id);
         this.setState({teams});
         this.setState({teamIds});
         this.setState({refreshTeams: false});
     }
 
-    openAddTeam = () => {
+    openAddTeam = (): void => {
         this.setState({addTeamOpen: true});
     }
 
-    addTeams = (teams) => {
+    addTeams = (teams: Array<any>): void => {
         const promises = [];
         for (const team of teams) {
             promises.push(this.props.actions.addUserToTeam(team.id, this.props.user.id));
         }
-        Promise.all(promises).finally(this.setState({refreshTeams: true}));
+        Promise.all(promises).finally(() => this.setState({refreshTeams: true}));
     }
 
-    closeAddTeam = () => {
+    closeAddTeam = (): void => {
         this.setState({addTeamOpen: false});
     }
 
-    doPasswordReset = (user) => {
+    doPasswordReset = (user: UserProfile): void => {
         this.setState({
             showPasswordModal: true,
             user,
         });
     }
 
-    doPasswordResetDismiss = () => {
+    doPasswordResetDismiss = (): void => {
         this.setState({
             showPasswordModal: false,
         });
     }
 
-    doPasswordResetSubmit = () => {
+    doPasswordResetSubmit = (): void => {
         this.setState({
             showPasswordModal: false,
         });
     }
 
-    handleMakeActive = (e) => {
+    handleMakeActive = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
         e.preventDefault();
         this.props.actions.updateUserActive(this.props.user.id, true).
             then(this.onUpdateActiveResult);
     }
 
-    handleShowDeactivateMemberModal = (e) => {
+    handleShowDeactivateMemberModal = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
         e.preventDefault();
         this.setState({showDeactivateMemberModal: true});
     }
 
-    handleDeactivateMember = () => {
+    handleDeactivateMember = (): void => {
         this.props.actions.updateUserActive(this.props.user.id, false).
             then(this.onUpdateActiveResult);
         this.setState({showDeactivateMemberModal: false});
     }
 
-    onUpdateActiveResult = ({error}) => {
+    onUpdateActiveResult = ({error}: any): void => {
```

Comment:
Does this have a type? It could be `ServerError` or something like that.

Reply:
`ActionResult` type has error set as `any` type,
By the previous code the state `error` was set to an empty string so I'm assuming this is a string.
Updated.

Reply:
Updated.